Geometry/CaloEventSetup/test/CaloAlignmentRcdRead.cc

```diff
@@ -18,32 +18,38 @@ class CaloAlignmentRcdRead : public edm::one::EDAnalyzer<>
 public:
 
   explicit CaloAlignmentRcdRead( const edm::ParameterSet& /*iConfig*/ )
-    :nEventCalls_(0) {}
+    :ebToken_{esConsumes<Alignments, EBAlignmentRcd>(edm::ESInputTag{})},
+     eeToken_{esConsumes<Alignments, EEAlignmentRcd>(edm::ESInputTag{})},
+     esToken_{esConsumes<Alignments, ESAlignmentRcd>(edm::ESInputTag{})},
+     nEventCalls_(0)
+  {}
   ~CaloAlignmentRcdRead() override {}
   
   template<typename T>
-  void dumpAlignments(const edm::EventSetup& evtSetup);
+  void dumpAlignments(const edm::EventSetup& evtSetup, edm::ESGetToken<Alignments, T>& token);
 
   void beginJob() override {}
   void analyze(edm::Event const& iEvent, edm::EventSetup const&) override;
   void endJob() override {}
   
 private:
+  edm::ESGetToken<Alignments, EBAlignmentRcd> ebToken_;
+  edm::ESGetToken<Alignments, EEAlignmentRcd> eeToken_;
+  edm::ESGetToken<Alignments, ESAlignmentRcd> esToken_;
 
   unsigned int nEventCalls_;
 };
 
 template<typename T>
-void CaloAlignmentRcdRead::dumpAlignments(const edm::EventSetup& evtSetup)
+void CaloAlignmentRcdRead::dumpAlignments(const edm::EventSetup& evtSetup, edm::ESGetToken<Alignments, T>& token)
 {
-  edm::ESHandle<Alignments> alignments;
-  evtSetup.get<T>().get(alignments);
+  const auto& alignments = evtSetup.getData(token);
   
   std::string recordName = Demangle(typeid(T).name())();
 
   LogDebug("CaloAlignmentRcdRead") << "Dumping alignments: " << recordName;
 
-  for (const auto & i : alignments->m_align) {
+  for (const auto & i : alignments.m_align) {
```

Comment:
```alignments.m_align``` looks odd. Shouldn't be ```alignments.align```?

Reply:
Well, that's what is in `Alignments` class
https://github.com/cms-sw/cmssw/blob/45b5000787f52ce1d75e3dc37c4cc69b9808dd8f/CondFormats/Alignment/interface/Alignments.h#L19